### PR TITLE
Fixed: Crash in Exchange Rates Screen

### PR DESCRIPTION
**Pull request by sambarboza**
# Description
- The app was crashing when typing an invalid currency code too fast in Exchange Rates Screen.

# Extra
- Added Espresso testing framework and Instrumentation Test for Exchange Rates currency search test case

# Evidence
### Espresso Test
- Running from command line in `dash-wallet/wallet` folder using command `./gradlew connected_testNet3DebugAndroidTest`

espresso

### CODE REVIEW

File: wallet/androidTest/testNet3/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (sambarboza) [inline wallet/androidTest/testNet3/res/values/strings.xml]: This file is only needed to generate the compiled resource file `R.txt` for Espresso tests, without it, test builds fail.

**Comment by HashEngineering** [comment]
Excellent.  I will review this very soon.

**Comment by HashEngineering** [comment]
The first commit appears to resolve the crash issue as I could not make it crash in my testing: 
 https://github.com/HashEngineering/dash-wallet/pull/70/commits/9e418b4d3b9e9af33891a8876f8fb33b397db69f

The other commits regarding adding Espresso testing - these results in crashing of the testnet Dash Wallet on my android 4.4.4 device.  I will post the stack trace separately.

File: wallet/build.gradle [hunk 2]
```diff
@@ -112,6 +129,7 @@ android {
         exclude 'org.bitcoin.test.checkpoints.txt'
         exclude 'messages.properties'
         exclude 'META-INF/services/sun.net.spi.nameservice.NameServiceDescriptor'
+        exclude 'META-INF/services/org.xmlpull.v1.XmlPullParserFactory'
```
Comment (HashEngineering) [inline wallet/build.gradle]: Does putting this in solve the issues when building?  I used to get about 39 or 40 errors from `xmlpull`

Comment (sambarboza) [reply wallet/build.gradle]: Yes, it stop warning about `xmlpull`.

**Comment by sambarboza** [comment]
The app was not compiling for Android 4.4 indeed.
Now tests are passing from Android 4.4 to 8.
I've removed Multidex for debug for now, we'll need some extra work to have it working on Android 4.4.

**Comment by HashEngineering** [comment]
I will review this code and test it within in the next day or two.

**Comment by HashEngineering** [comment]
Everything works.  I will be merging soon.

This was the warnings or errors that I was getting that I thought would be solved with
`exclude 'META-INF/services/org.xmlpull.v1.XmlPullParserFactory'`

However, my errors are from org.xbill...

```
AGPBI: {"kind":"error","text":"warning: Ignoring InnerClasses attribute for an anonymous inner class","sources":[{}]}
AGPBI: {"kind":"error","text":"(org.xbill.DNS.APLRecord$1) that doesn\u0027t come with an","sources":[{}]}
AGPBI: {"kind":"error","text":"associated EnclosingMethod attribute. This class was probably produced by a","sources":[{}]}
AGPBI: {"kind":"error","text":"compiler that did not target the modern .class file format. The recommended","sources":[{}]}
AGPBI: {"kind":"error","text":"solution is to recompile the class from source, using an up-to-date compiler","sources":[{}]}
AGPBI: {"kind":"error","text":"and without specifying any \"-target\" type options. The consequence of ignoring","sources":[{}]}
AGPBI: {"kind":"error","text":"this warning is that reflective operations on this class will incorrectly","sources":[{}]}
AGPBI: {"kind":"error","text":"indicate that it is *not* an inner class.","sources":[{}]}
AGPBI: {"kind":"error","text":"warning: Ignoring InnerClasses attribute for an anonymous inner class","sources":[{}]}
AGPBI: {"kind":"error","text":"(org.xbill.DNS.Compression$1) that doesn\u0027t come with an","sources":[{}]}
AGPBI: {"kind":"error","text":"associated EnclosingMethod attribute. This class was probably produced by a","sources":[{}]}
AGPBI: {"kind":"error","text":"compiler that did not target the modern .class file format. The recommended","sources":[{}]}
AGPBI: {"kind":"error","text":"solution is to recompile the class from source, using an up-to-date compiler","sources":[{}]}
AGPBI: {"kind":"error","text":"and without specifying any \"-target\" type options. The consequence of ignoring","sources":[{}]}
AGPBI: {"kind":"error","text":"this warning is that reflective operations on this class will incorrectly","sources":[{}]}
AGPBI: {"kind":"error","text":"indicate that it is *not* an inner class.","sources":[{}]}
AGPBI: {"kind":"error","text":"warning: Ignoring InnerClasses attribute for an anonymous inner class","sources":[{}]}
AGPBI: {"kind":"error","text":"(org.xbill.DNS.Tokenizer$1) that doesn\u0027t come with an","sources":[{}]}
AGPBI: {"kind":"error","text":"associated EnclosingMethod attribute. This class was probably produced by a","sources":[{}]}
AGPBI: {"kind":"error","text":"compiler that did not target the modern .class file format. The recommended","sources":[{}]}
AGPBI: {"kind":"error","text":"solution is to recompile the class from source, using an up-to-date compiler","sources":[{}]}
AGPBI: {"kind":"error","text":"and without specifying any \"-target\" type options. The consequence of ignoring","sources":[{}]}
AGPBI: {"kind":"error","text":"this warning is that reflective operations on this class will incorrectly","sources":[{}]}
AGPBI: {"kind":"error","text":"indicate that it is *not* an inner class.","sources":[{}]}
AGPBI: {"kind":"error","text":"warning: Ignoring InnerClasses attribute for an anonymous inner class","sources":[{}]}
AGPBI: {"kind":"error","text":"(org.xbill.DNS.UDPClient$1) that doesn\u0027t come with an","sources":[{}]}
AGPBI: {"kind":"error","text":"associated EnclosingMethod attribute. This class was probably produced by a","sources":[{}]}
AGPBI: {"kind":"error","text":"compiler that did not target the modern .class file format. The recommended","sources":[{}]}
AGPBI: {"kind":"error","text":"solution is to recompile the class from source, using an up-to-date compiler","sources":[{}]}
AGPBI: {"kind":"error","text":"and without specifying any \"-target\" type options. The consequence of ignoring","sources":[{}]}
AGPBI: {"kind":"error","text":"this warning is that reflective operations on this class will incorrectly","sources":[{}]}
AGPBI: {"kind":"error","text":"indicate that it is *not* an inner class.","sources":[{}]}
AGPBI: {"kind":"error","text":"warning: Ignoring InnerClasses attribute for an anonymous inner class","sources":[{}]}
AGPBI: {"kind":"error","text":"(org.xbill.DNS.ZoneTransferIn$1) that doesn\u0027t come with an","sources":[{}]}
AGPBI: {"kind":"error","text":"associated EnclosingMethod attribute. This class was probably produced by a","sources":[{}]}
AGPBI: {"kind":"error","text":"compiler that did not target the modern .class file format. The recommended","sources":[{}]}
AGPBI: {"kind":"error","text":"solution is to recompile the class from source, using an up-to-date compiler","sources":[{}]}
AGPBI: {"kind":"error","text":"and without specifying any \"-target\" type options. The consequence of ignoring","sources":[{}]}
AGPBI: {"kind":"error","text":"this warning is that reflective operations on this class will incorrectly","sources":[{}]}
AGPBI: {"kind":"error","text":"indicate that it is *not* an inner class.","sources":[{}]}
```

**Comment by HashEngineering** [comment]
Mistakenly pressed close.

File: wallet/build.gradle (hunk 2 above)
Comment (sambarboza) [reply wallet/build.gradle]: Actually, as you just mentioned below, it doesn't stop to warn about `xmlpull`, it prevents a compiling error.

**Comment by HashEngineering** [comment]
This was rebased and then the Expresso commits were squash into one.

This has been merged as:  https://github.com/HashEngineering/dash-wallet/commit/9f5b7ab84b185d411ac95c0333b41ea267760cbf and https://github.com/HashEngineering/dash-wallet/commit/30bb304c03ece1b749fc90aad896f7af64e0fb36

**Pull request closed without merging**